src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -573,85 +573,64 @@ class spell_pri_mind_bomb : public AuraScript
 };
 
 // 47540 - Penance
-class spell_pri_penance : public SpellScriptLoader
+class spell_pri_penance : public SpellScript
 {
-    public:
-        spell_pri_penance() : SpellScriptLoader("spell_pri_penance") { }
-
-        class spell_pri_penance_SpellScript : public SpellScript
-        {
-            PrepareSpellScript(spell_pri_penance_SpellScript);
-
-            bool Load() override
-            {
-                return GetCaster()->GetTypeId() == TYPEID_PLAYER;
-            }
+    PrepareSpellScript(spell_pri_penance);
 
-            bool Validate(SpellInfo const* spellInfo) override
-            {
-                SpellInfo const* firstRankSpellInfo = sSpellMgr->GetSpellInfo(SPELL_PRIEST_PENANCE_R1, DIFFICULTY_NONE);
-                if (!firstRankSpellInfo)
-                    return false;
-
-                // can't use other spell than this penance due to spell_ranks dependency
-                if (!spellInfo->IsRankOf(firstRankSpellInfo))
-                    return false;
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_PENANCE_DAMAGE,
+            SPELL_PRIEST_PENANCE_HEALING
+        });
+    }
 
-                uint8 rank = spellInfo->GetRank();
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_DAMAGE, rank, true))
-                    return false;
-                if (!sSpellMgr->GetSpellWithRank(SPELL_PRIEST_PENANCE_R1_HEAL, rank, true))
-                    return false;
+    bool Load() override
+    {
+        return GetCaster()->GetTypeId() == TYPEID_PLAYER;
+    }
 
-                return true;
-            }
+    SpellCastResult CheckCast()
+    {
+        Player* caster = GetCaster()->ToPlayer();
```

Comment:
`Player*` stuff is not used, you can simply use `GetCaster()`

Reply:
It is used to check !caster->IsFriendlyTo(target).

Reply:
Definition of `IsFriendlyTo` is in `WorldObject`, so we do not require `Player*`

Reply:
ehh that makes sense, let me change it then